### PR TITLE
Allow defining custom JSON field names for DTO interfaces

### PR DESCRIPTION
**Pull request by tareqhs**
A new annotation, \@JsonFieldName, allow defining it

Signed-off-by: Tareq Sharafy <tareq.sharafy@sap.com>

### CODE REVIEW

File: platform-api/che-core-api-dto/src/main/java/org/eclipse/che/dto/generator/DtoImpl.java [hunk 1]
```diff
@@ -324,6 +349,23 @@ protected boolean isLastMethod(Method method) {
         Preconditions.checkNotNull(method);
         return method == dtoMethods.get(dtoMethods.size() - 1);
     }
+    
+    /**
+     * Create a textual representation of a string literal that evaluates to the given value.
+     */
+    protected String quoteStringLiteral(String value) {
+        StringWriter sw = new StringWriter();
+        JsonWriter writer = new JsonWriter(sw);
```
Comment (mshaposhnik) [inline platform-api/che-core-api-dto/src/main/java/org/eclipse/che/dto/generator/DtoImpl.java]: may we use try-with-resources here?

**Comment by skabashnyuk** [comment]
Add @Beta on JsonFieldName

File: platform-api/che-core-api-dto/src/main/java/org/eclipse/che/dto/generator/DtoImpl.java (hunk 1 above)
Comment (tareqhs) [inline platform-api/che-core-api-dto/src/main/java/org/eclipse/che/dto/generator/DtoImpl.java]: @mshaposhnik done

**Comment by tareqhs** [comment]
@skabashnyuk done

**Pull request merged**